test/unit/core/tag/lifecycle.ts

```diff
@@ -46,4 +47,84 @@ suite('Lifecycle', ({ expect, spy, stub }) => {
       expect(dispatch).to.be.calledWith(action2);
     });
   });
+
+  describe('onInitialize()', () => {
+    it('should add sugar methods, metadata and actions', () => {
+      const tag = { a: 'b', on: () => null };
+      const addSugar = stub(Lifecycle, 'addSugar');
+      const addMetadata = stub(Lifecycle, 'addMetadata');
+      const primeTagActions = stub(Lifecycle, 'primeTagActions');
+      stub(Lifecycle, 'onRecalculateProps');
+
+      Lifecycle.onInitialize.call(tag);
+
+      expect(addSugar).to.be.calledWith(tag);
+      expect(addMetadata).to.be.calledWith(tag);
+      expect(primeTagActions).to.be.calledWith(tag);
+    });
+
+    it('should set initial props', () => {
+      const tag = { a: 'b', on: () => null };
+      const onRecalculateProps = stub(Lifecycle, 'onRecalculateProps');
+      stub(Lifecycle, 'addSugar');
+      stub(Lifecycle, 'addMetadata');
+      stub(Lifecycle, 'primeTagActions');
+
+      Lifecycle.onInitialize.call(tag);
+
+      expect(onRecalculateProps.thisValues[0]).to.eq(tag);
+    });
+
+    it('should listen for BEFORE_MOUNT', () => {
+      const on = spy();
+      const trigger = spy();
+      const tag = { a: 'b', on, trigger };
+      const onRecalculateProps = stub(Lifecycle, 'onRecalculateProps');
+      stub(Lifecycle, 'addSugar');
+      stub(Lifecycle, 'addMetadata');
+      stub(Lifecycle, 'primeTagActions');
+
+      Lifecycle.onInitialize.call(tag);
+
+      expect(on).to.be.calledWith(Lifecycle.Phase.BEFORE_MOUNT, sinon.match((cb) => {
+        cb();
+
+        return expect(trigger).to.be.calledWith(Lifecycle.Phase.STATE_FINALIZED);
+      }));
+    });
+  });
+
+  describe('onBeforeMount()', () => {
+    it('should apply storefront gb-stylish class', () => {
+      const add = spy();
+      const tag = { root: { classList: { add, remove: () => null } }, props: { stylish: true } };
+
+      Lifecycle.onBeforeMount.call(tag);
+
+      expect(add).to.be.calledWith(STYLISH_CLASS);
```

Comment:
Can't you use calledWithExactly instead of checking that its called with this and not called with UI_CLASS?